src/components/eip4626/user/VaultDepositWithdraw.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  Box,
+  Text,
+  Flex,
+  Input,
+  Tabs,
+  TabList,
+  TabPanels,
+  Tab,
+  TabPanel,
+  Button,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import Icon from "@/components/Icon";
+import { useMemo } from "react";
+import { useDepositWithdrawFlow } from "@/hooks/useDepositWithdrawFlow";
+import { EvmAddress } from "@/types/types";
+import { VaultActionResults } from "@/components/eip4626/user/VaultActionResults";
+
+type Props = {
+  vaultAddress: EvmAddress;
+  toggleError: (isError: boolean) => void;
+  error: boolean;
+};
+
+export const VaultDepositWithdraw = ({
+  vaultAddress,
+  toggleError,
+  error,
+}: Props) => {
+  const {
+    withdrawForm,
+    depositForm,
+    depositError,
+    approveError,
+    depositResult,
+    approveResult,
+    shareUserBalance,
+    vaultAssetUserBalance,
+    approveToken,
+  } = useDepositWithdrawFlow(vaultAddress);
```

Comment:
This hook does not look like a hook anymore, considering the amount of things its exporting.
It violates couple of principles, like "single responsibility" one, also I think it's not that pure anymore

https://react.dev/reference/rules/components-and-hooks-must-be-pure

I would suggest to consider splitting it into smaller components/hooks, alongside the other suggestions in this PR about the structure.